Source/JavaScriptCore/dfg/DFGConstantFoldingPhase.cpp

```diff
@@ -417,7 +417,7 @@ class ConstantFoldingPhase : public Phase {
             case GetArrayLength: {
                 ArrayMode arrayMode = node->arrayMode();
                 AbstractValue& abstractValue = m_state.forNode(node->child1());
-                if (arrayMode.type() != Array::AnyTypedArray && arrayMode.isSomeTypedArrayView() && !arrayMode.mayBeResizableOrGrowableSharedTypedArray()) {
+                if (arrayMode.type() != Array::AnyTypedArray && arrayMode.isSomeTypedArrayViewOrDataView() && !arrayMode.mayBeResizableOrGrowableSharedTypedArray()) {
```

Comment:
I think this should not include DataView since `DataView.length` does not exist.